gobblin-runtime/src/main/java/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -484,9 +521,26 @@ private void unlockJob() {
    * {@link JobListeners#parallelJobListener(List)} to create a {@link CloseableJobListener} that will execute all
    * the {@link JobListener}s in parallel.
    */
-  private CloseableJobListener getParallelCombinedJobListener(JobListener jobListener) {
+  private CloseableJobListener getParallelCombinedJobListener(JobState jobState, JobListener jobListener) {
     List<JobListener> jobListeners = Lists.newArrayList(this.mandatoryJobListeners);
     jobListeners.add(jobListener);
+
+    Set<String> jobListenerClassNames = jobState.getPropAsSet(ConfigurationKeys.JOB_LISTENERS_KEY, StringUtils.EMPTY);
+    if (jobListenerClassNames != null) {
+      for (String jobListenerClassName : jobListenerClassNames) {
+          try {
+            @SuppressWarnings("unchecked")
+            Class<? extends JobListener> jobListenerClass =
+                    (Class<? extends JobListener>) Class.forName(jobListenerClassName);
+            Constructor<? extends JobListener> jobListenerConstructor = jobListenerClass.getConstructor();
```

Comment:
I was using a similar method to how DataPublishers where instantiated.  I can switch this to newInstance.